packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,303 @@
+
+### DOM
+
+```html
+<div role="dialog" class="fui-dialog">{children}</div>
+```
+
+## Dialog Header
+
+The Header component will expect to have a dialog title/header and will show by default the close (X icon) button.
+
+### DOM
+
+```html
+<div class="fui-dialog-header">
+  <span>{title}</span>
+  <button aria-label="close" />
+</div>
+```
+
+### API
 
-_Provide some representative example code that uses the proposed API for the component_
+| Property          | Values  | Default     | Purpose                         |
+| ----------------- | ------- | ----------- | ------------------------------- |
+| `closeButton`     | _slot_  | `undefined` | Button that closes the dialog   |
+| `hideCloseButton` | boolean | `false`     | Shows or hides the close button |
 
-## Variants
+## Dialog Body
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+The body is a container where the content of the dialog is rendered. Apart from padding, this component does not have other behaviour.
 
-## API
+```html
+<div class="fui-dialog-body">{children}</div>
+```
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+## Dialog Footer
 
-## Structure
+The footer is a container for the actions of the dialog, which must be not more than 3 (primary, secondary and tertiary actions).
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### DOM
+
+```html
+<div class="fui-dialog-footer">
+  <button>{tertiaryAction}</button>
+  <button>{primaryAction}</button>
+  <button>{secondaryAction}</button>
+</div>
+```
+
+> ⚠️ _Pending issue: should there be any handling for the order of buttons? This also, includes the focus sequence of the buttons when the dialog has a tertiary button as well._
+
+## Sample Code
+
+### App
+
+```tsx
+<Button onClick={() => setIsOpen(true)} aria-haspopup={true}>
+  Open Dialog
+</Button>
+
+<Dialog open={isOpen}>
+  <DialogHeader>Dialog title</DialogHeader>
+  <DialogBody>Dialog's main content</DialogBody>
+  <DialogFooter>
+    <Button>Submit</Button>
+    <Button>Cancel</Button>
+  </DialogFooter>
+</Dialog>
+```
+
+### DOM structure
+
+```html
+<button>Open Dialog</button>
+<!-- Something something something -->
+<div class="fui-portal">
+  <div role="dialog" class="fui-dialog">
+    <div class="fui-dialog-overlay" />
+    <header class="fui-dialog-header">Missing Subject</header>
+    <div class="fui-dialog-body">Do you want to send this message without a subject?</div>
+    <footer class="fui-dialog-footer">
+      <button>Send</button>
+      <button>Cancel<button>
+    </footer>
+  </div>
+</div>
+```
 
 ## Migration
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+_TBA: Link to migration guide doc_
+
+## Behaviours
+
+### Mouse & touch
+
+#### Modal
```

Comment:
Would you like to mention scroll behavior? It's a common pattern block scroll on overlays, for example with [`react-remove-scroll`](https://www.npmjs.com/package/react-remove-scroll)

ARIA example does it, https://www.w3.org/TR/wai-aria-practices-1.2/examples/dialog-modal/dialog.html

Reply:
Yes, so the `modal` and `alert` dialog do definitely need the scroll block of the body element. I did not add it as I thought it was a detail, but I'll update the spec to mention this. 
Thanks for pointing to resources ^_^